### PR TITLE
Detect operationsApi/http port collisions during config validation

### LINKED ISSUE #1412: Secure-port (TLS) WebSocket upgrades hang / never complete on 5.1.x (works on insecure port) — regression since 5.1.0
### Bug Summary

WebSocket upgrades on the **secure (TLS) HTTP port** (`http.securePort`) hang and never complete on Harper 5.1.x — the upgrade request reaches the server but no response is ever sent, and the client eventually fails with `socket hang up`. The **same WebSocket on the insecure HTTP port** (`http.port`) upgrades correctly (`101 Switching Protocols`), and **plain HTTPS** on the secure port works fine (`200`). So only the TLS-terminated WebSocket *upgrade* path is broken.

Bisected to the **5.1.0 middleware/routing rewrite**. Confirmed on 5.1.5.

### Steps to Reproduce

1. Run Harper 5.1.x configured with both an insecure and a secure HTTP port (`http.port` + `http.securePort`, `http2: false`) and a Resource that exposes a WebSocket (e.g. a table subscription resource).
2. Open a WebSocket to the **insecure** port — succeeds:
   ```
   curl -s -i -H 'Connection: Upgrade' -H 'Upgrade: websocket' \
     -H 'Sec-WebSocket-Version: 13' -H 'Sec-WebSocket-Key: dGhlIHNhbXBsZSBub25jZQ==' \
     http://localhost:9980/<resource>/<id>
   → HTTP/1.1 101 Switching Protocols
   ```
3. Open the **same** WebSocket to the **secure** port — hangs, no response:
   ```
   curl -sk -i -H 'Connection: Upgrade' -H 'Upgrade: websocket' \
     -H 'Sec-WebSocket-Version: 13' -H 'Sec-WebSocket-Key: dGhlIHNhbXBsZSBub25jZQ==' \
     https://localhost:9926/<resource>/<id>
   → (no response; connection hangs until timeout / "socket hang up")
   ```
4. For contrast, a plain HTTPS GET on the secure port returns normally:
   ```
   curl -sk -o /dev/null -w '%{http_code}' https://localhost:9926/   → 200
   ```

These were run **directly against the container's localhost ports** (no reverse proxy or port-forwarder in the path), so the hang is in Harper itself.

### Expected Behavior

A WebSocket upgrade on the secure port returns `101 Switching Protocols` and establishes the connection, identical to the insecure port.

### Actual Behavior

The secure-port upgrade request is received but never answered — no `101`, no error — and the socket hangs until the client times out (`socket hang up`). Plain HTTPS on the same port is unaffected. Reproduces on every worker.

### Platform

harper-pro 5.1.5 (Harper core 5.1.x), `http2: false`, Linux / Docker. Regression present since core 5.1.0; the affected instance previously ran 5.0.30 where secure-port WS worked.

### Console Errors

Client-side:
```
Error: socket hang up
Request URL: https://<host>/<resource>/<id>
Request Method: GET
Sec-WebSocket-Version: 13
Connection: Upgrade
Upgrade: websocket
```
Server logs show nothing for the failed upgrade attempt.

### Additional Context

**Ruled out:**
- **Not a reverse-proxy / infra issue** — reproduced inside the container against `localhost`, with all proxies removed from the path.
- **Not HTTP/2** — effective config is `http2: false`, and the secure port negotiates `http/1.1` via ALPN even when the client offers `h2`. So this is the HTTP/1.1 upgrade path, not an HTTP/2-doesn't-do-Upgrade situation.

**Bisection (suspect area, not yet line-pinned):**
- The instance was upgraded 5.0.30 → 5.1.5; secure-port WS worked on 5.0.30.
- 5.1.0 introduced the new middleware/routing system that explicitly covers `handleHttpRequest` / `onWebSocket` / `onUpgrade`.
- Suspect changes in `server/http.ts` + `server/middlewareChain.ts`: `62b9c983e` ("Extract middleware chain logic"), `cb550a3a7` ("Fix dispatch signature for ws and upgrade chains").
- The `onWebSocket` upgrade wiring itself (`server.on('upgrade')` → `upgradeChains[port]` → `websocketServers[port].handleUpgrade`) is **structurally unchanged** from 5.0.30, so the defect is most likely in the rewritten chain machinery or how the **secure** port's upgrade chain is built/dispatched (it works for the insecure port using the same wiring). Pinning the exact line needs a live trace (logpoint in the upgrade chain on a secure vs. insecure worker).

**Test gap:** the secure-port (wss) upgrade path appears uncovered — insecure-port WS is exercised and passes. A regression test that drives a `wss://` upgrade against `securePort` would catch this and guard against recurrence.

**Workaround:** terminate TLS at a front proxy and forward the (plaintext) WebSocket to Harper's insecure `http.port`, where upgrades work.

### Are you planning to fix this issue?

Maybe, depending on complexity

### PR DESCRIPTION
**Pull request by kriszyp**
Fixes #1412.

## Summary
Adds a cross-section port-collision check to `validateConfig` (`config/configUtils.js`) so a config where the operations-API and REST/http servers are pointed at the same port fails loudly at startup instead of silently breaking the secure-port WebSocket path.

## Why
Issue #1412 reports secure-port (`wss`) WebSocket upgrades hanging on 5.1.x while the insecure port works. Root cause (confirmed by live + local repro on the issue) is a **config port collision**, not the 5.1.0 middleware rewrite it first looked like:

- In 5.1 the operations API binds on the **main thread without `SO_REUSEPORT`**; the REST/http servers bind on the **workers with it**.
- When a config collides another server onto the REST `securePort` (e.g. `operationsApi.network.securePort` == `http.securePort`), the single non-reusePort ops socket claims the port **exclusively**. The reusePort REST workers then hit `EADDRINUSE`, which is swallowed in `threadServer.js` (that swallow is load-bearing on macOS/Windows, where every server is non-reusePort, so it can't simply be made loud).
- The lone surviving listener is the ops server, which has no REST WebSocket upgrade handler → `wss` upgrades reset/hang. 5.0 didn't break because the ops API also used reusePort.

`validateConfig` already hard-errors on *within-section* collisions (`http.port === http.securePort`, same for ops). This just closes the missing *cross-section* case at the right layer.

## Where to put attention
- **Behavior change (deliberate):** this is a hard error, consistent with the existing within-section checks. A deployment currently running with this collision (one server silently shadowed) will now **refuse to start** until the config is fixed. That's the intended fail-loud trade vs. silent WS breakage, but it's the thing to weigh — especially re: whether this should be cherry-picked to 5.1 (happy to, via /patch-pr, if we want it released; see open question below).
- **Scope of the check** (`config/configUtils.js`): limited to the `http` ↔ `operationsApi.network` pairing on purpose. `mqtt.websocket` *intentionally* shares `http.port`, and `replication.port` defaults to the operations port, so a blanket "all ports distinct" check would false-positive. Worth a sanity check that this scoping matches your mental model of which servers can legitimately co-bind.
- Ports are normalized via `Number()` (string env-var port vs numeric YAML port) and `0`/ephemeral is skipped — see the two edge-case unit tests.

## Not in this PR (open follow-ups)
- The **`wss`-upgrade test gap** the issue flags (a regression test driving a TLS WebSocket upgrade against `securePort`) is real but orthogonal to this config-layer fix and needs net-new integration-test infrastructure — recommend a separate PR.
- Detecting a **component-declared `http.port`** colliding with the REST secure port (the 4th repro permutation in the issue) — components load after main-config validation, so it'd need a different hook; deferred.

## Open question
Should this be cherry-picked to the 5.1 release branch as a patch? It's safe and well-contained, but it's a startup-refusal behavior change (see above).

🤖 Generated by Claude (Opus 4.8). Cross-model review: Gemini (`agy`) leg run; two edge cases it surfaced (string-vs-number port equality, port-0 ephemeral false-positive) are fixed in this PR with tests. Codex leg unavailable in this environment.

### CODE REVIEW

File: config/configUtils.js [hunk 1]
```diff
@@ -466,6 +466,40 @@ function validateConfig(configDoc, skipFsValidation = false) {
...
+		['operationsApi.network.port', configJson.operationsApi?.network?.port],
+		['operationsApi.network.securePort', configJson.operationsApi?.network?.securePort],
+	];
+	const portLabels = new Map();
+	for (const [label, value] of configuredPorts) {
+		// Normalize to a number so a string env-var port (e.g. "9926") still matches a numeric YAML port, and skip
+		// unset/ephemeral values: 0 (and NaN from a non-numeric value) requests an OS-assigned port, so distinct
+		// sections configured that way won't actually collide at bind time.
+		const port = Number(value);
+		if (!port) continue;
```
[the remark this answers is not in the document]
Comment (kriszyp) [reply config/configUtils.js]: Good catch — adopted in 7a04109f9. Switched the guard to the already-imported `is-number` helper so a malformed boolean/array/object is skipped (and reported by the schema validator) rather than coerced into a spurious port. Numeric strings still pass, so a string env-var port continues to match a numeric YAML port. Added a unit test covering the non-numeric case. Thanks for the review! — Claude (Opus 4.8)
